setuptools/tests/test_ccompiler.py

```diff
@@ -0,0 +1,8 @@
+from setuptools.ccompiler import CCompiler
+from setuptools.ccompiler import customize_compiler
+from setuptools.ccompiler import new_compiler
+
+def test_ccompiler_namespace():
+    ccompiler = new_compiler()
+    customize_compiler(ccompiler)
+    assert hasattr(ccompiler, "compile")
```

Comment:
```suggestion
    assert hasattr(ccompiler, "compile")


@pytest.fixture
def _avoid_permanent_changes_in_sysconfig(monkeypatch):
    import importlib
    import sys

    # Avoid caching `distutils.sysconfig` and force it to be re-imported later.
    # This should "cancel out" any permanent changes that comes as a side-effect of
    # import thing `distutils.sysconfig`.
    monkeypatch.setattr(sys, "modules", sys.modules.copy())
    yield
    importlib.invalidate_caches()
```

This is an attempt to "undo" the `import distutils.sysconfig`.

I am not entirely sure about using `importlib.invalidate_caches()`, it might be a double edged sword... I am also not sure that it makes a difference for the default import finders in CPython...

Reply:
Thank you so much for your help ! I will study your suggestion before I commit it, because for now it's black magic to me, but I really appreciate it !